### ISSUE TITLE
Semaphore state after vkAcquireNextImageKHR fails with VK_ERROR_OUT_OF_DATE_KHR

### ISSUE THREAD

**Issue by Overv** [KhronosGroup/Vulkan-Docs#1126]
The description of `vkQueuePresentKHR` clearly states what happens to synchronization operations, but the one for `vkAcquireNextImageKHR` does not.

If I call `vkAcquireNextImageKHR` and pass a semaphore then what is the state of that semaphore if the call returns `VK_ERROR_OUT_OF_DATE_KHR`?

Will it be unchanged (unsignaled and not waiting), transition to waiting forever or wait and very quickly be signaled?

**Comment by oddhack** [comment]
Assigned to SI TSG for triage.

**Comment by stonesthrow** [comment]
We will investigate our drivers, but expectation is there is no change to the Semaphore under this condition. Resolution and text in process.

**Comment by krOoze** [comment]
Not just `VK_ERROR_OUT_OF_DATE_KHR`.

On one hand the spec says:

> If `semaphore` is not `VK_NULL_HANDLE`, the application **may** assume that, once `vkAcquireNextImageKHR` returns, the semaphore signal operation referenced by `semaphore` has been submitted for execution.

On the other hand, it feels riddiculous the `semaphore` would still be scheduled for execution even if `VK_NOT_READY` is returned.

**Comment by Tobski** [comment]
> 
> 
> Not just `VK_ERROR_OUT_OF_DATE_KHR`.
> 
> On one hand the spec says:
> 
> > If `semaphore` is not `VK_NULL_HANDLE`, the application **may** assume that, once `vkAcquireNextImageKHR` returns, the semaphore signal operation referenced by `semaphore` has been submitted for execution.
> 
> On the other hand, it feels riddiculous the `semaphore` would still be scheduled for execution even if `VK_NOT_READY` is returned.

I wrote that sentence originally and we intended for it to be under the context of the start of that paragraph where it says "When successful, ..." but reading it now I have no idea why I thought that was clear :P

**Comment by chadversary** [comment]
Update. This is tracked by internal issue 1923, which is near completion.

**Comment by krOoze** [comment]
@oddhack 2009 (which was closed right now) was dup\superset of this, so this should probably close too.

**Comment by stonesthrow** [comment]
Agreed, the new change specifies the status of fence and semaphore of a ANI depending on return. I think this can be closed
